server/src/core/odoo.rs

```diff
@@ -926,146 +920,68 @@ impl Odoo {
-            if let Some(odoo_path) = odoo_path {
-                config.odoo_path = Some(odoo_path.as_str().expect("odooPath must be a String").to_string());
-            }
-            if let Some(python_path) = odoo_conf.get("finalPythonPath") {
-                if python_path.is_string() {
-                    config.python_path = python_path.as_str().unwrap().to_string();
-                } else {
-                    session.log_message(MessageType::ERROR, String::from("pythonPath must be a string, using 'python' as pythonPath"));
-                    config.python_path = S!("python");
-                }
-            } else {
-                session.log_message(MessageType::ERROR, String::from("pythonPath must be defined, using 'python' as pythonPath"));
-                config.python_path = S!("python");
-            }
-            if let Some(additional_stubs) = odoo_conf.get("additional_stubs"){
-                config.additional_stubs = additional_stubs.as_array().expect("additional_stubs must be an Array")
-                .into_iter().map(|v| v.as_str().expect("additional_stubs values must be strings").to_string()).collect();
-            }
-        } else {
-            config.addons = vec![];
-            config.odoo_path = None;
-            session.log_message(MessageType::ERROR, S!("Unable to find selected configuration. No odoo path has been found."));
-        }
-        config.file_cache = _file_cache;
-        config.refresh_mode = _refresh_mode;
-        config.auto_save_delay = _auto_save_delay;
-        config.ac_filter_model_names = _ac_filter_model_names;
-        config.diag_missing_imports = _diag_missing_imports;
-
-        debug!("Final config: {:?}", config);
-        Ok(config)
     }
 
     pub fn init(session: &mut SessionInfo) {
         let start = std::time::Instant::now();
         session.log_message(MessageType::LOG, String::from("Building new Odoo knowledge database"));
-        let config = Odoo::update_configuration(session);
+        let config = get_configuration(session.sync_odoo.get_file_mgr().borrow().get_workspace_folders());
+        if let Ok((_, config_file)) = &config {
+            session.sync_odoo.config_file = Some(config_file.clone());
+            Odoo::send_all_configurations(session);
+        }
+        let maybe_selected_config = match Odoo::read_selected_configuration(session){
+            Ok(config) => config,
+            Err(e) => {
+                session.show_message(MessageType::ERROR, format!("Unable to read selected configuration: {}  \n\nPlease select a correct profile or fix the issues in the config", e));
+                error!(e);
+                return;
+            }
+        };
+        let selected_config = match maybe_selected_config {
+            None => {
+                session.show_message(MessageType::INFO, String::from("No Odoo configuration selected. Please select a configuration in the settings."));
```

Comment:
Shouldn't we take the first by default if the setting doesn't exist in the config? 
Else any IDE implementation that doesn't implement the config switch interface won't be able to use it?
I don't know